charts/skypilot/templates/api-deployment.yaml

```diff
@@ -108,6 +108,20 @@ spec:
         - name: SKYPILOT_AUTH_USER_HEADER
           value: {{ .Values.apiService.authUserHeaderName | quote }}
         {{- end }}
+        {{- if .Values.awsCredentials.enabled }}
+        - name: AWS_ACCESS_KEY_ID
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.awsCredentials.awsSecretName }}
+              key: {{ .Values.awsCredentials.accessKeyIdKeyName }}
+              optional: true
+        - name: AWS_SECRET_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.awsCredentials.awsSecretName }}
+              key: {{ .Values.awsCredentials.secretAccessKeyKeyName }}
+              optional: true
+        {{- end }}
```

Comment:
Seems like we don't need this in this container since it will be written to ~/.aws/credentials in the initContainer